### PR TITLE
Add documentation for webactions and enhance production tracker docs

### LINKED ISSUE #252: Add Documentation for webactions
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

.

### How would you imagine the implementation of the feature?

maybe we can have:
- artist guide about how to use the web actions and explore the capabilities and limitations.
- admin guide
- Dev guide how to add your custom web actions

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

### Describe alternatives you've considered:

_No response_

### Additional context:

_No response_

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
resolve #252 
This PR adds:
- [x] Details Panel doc: that explains the details panel content. main reason to add it, is to write about web actions.
- [x] web actions artist doc: it's added in details panel.
- [x] web actions admin doc: Admins can't control/customize web actions. therefore, I only added a note about web actions at the top of application addon admin doc.
- [x] web actions developer docs

## Testing notes:
1. Read through changes
2. Approve they make sense to you.

### CODE REVIEW

File: website/docs/artist_details_panel.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+![](assets/details_panel/details_panel_overview.png)
+
+- **Folder Path**: Displays the folder path of the current task.
+- **Name & Thumbnail**: Shows the name and thumbnail of the task.
+:::tip Update Thumbnail
+You can update the thumbnail by right-clicking the thumbnail and selecting Upload new thumbnail.
+:::
+- **Tags**: View and add tags.
+- **Status**: View and update status.
+- **Web Actions**: Access available web actions to launch applications directly via the AYON web server.
```
Comment (iLLiCiTiT) [inline website/docs/artist_details_panel.md:27]:
```suggestion
- **Web Actions**: Access available web actions, e.g. applications, directly via the AYON web server.
```

File: website/docs/addon_applications_admin.md [hunk 2]
```diff
@@ -15,7 +15,10 @@ import versions from '@site/docs/assets/json/Ayon_addons_version.json'
 {versions.Applications_Badge}
 </ReactMarkdown>
 
-## Introduction
+
+## Applications Addon Features
+
+### Applications, Tools and Filters
 
 The Applications addon streamlines the setup of applications and tools across your studio. It empowers you to set executable paths and startup environments, with support for various operating systems including Windows, Linux, and macOS.
```
Comment (Innders) [inline website/docs/addon_applications_admin.md:23]: empowers is the wrong word. Use "allows" instead.

File: website/docs/addon_applications_admin.md [hunk 3]
```diff
@@ -30,6 +33,20 @@ The Addon features two primary settings categories: Definitions and Filters:
...
+### Applications Web Actions
+
+:::info
+The web actions feature requires AYON Server version `1.3.0` or higher, core addon version `0.4.4` or higher, and Launcher version `1.1.0` or higher.
+:::
+
+Web actions let you launch applications directly through the AYON web server. You can find these actions in the [details panel](artist_details_panel.md).
+
+The application addon handles the logic for web actions.Currently, there are no specific settings for web actions.
+
```
Comment (Innders) [inline website/docs/addon_applications_admin.md:46]:
```suggestion
The application addon handles the logic for web actions. Currently, there are no specific settings for web actions.
```

File: website/docs/addon_applications_admin.md [hunk 4]
```diff
@@ -30,6 +33,20 @@ The Addon features two primary settings categories: Definitions and Filters:
...
+Remember, web actions are generated from your application's definitions in your production or development bundles, based on the developer mode setting. They also follow your application's filter settings.
+
```
Comment (Innders) [inline website/docs/addon_applications_admin.md:48]:
```suggestion
Web actions are generated from your application's definitions in your production or development bundles, depending on the developer mode setting, and they adhere to your application’s filter settings.
```

File: website/docs/artist_details_panel.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+id: artist_details_panel
+title: The Task Details Panel
+sidebar_label: Details Panel
+---
+
+:::info
+The details panels received updates in server version `1.3.0`.
+:::
```
Comment (Innders) [inline website/docs/artist_details_panel.md:9]: I don't think this is necessary and can be removed.

File: website/docs/artist_details_panel.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- **Name & Thumbnail**: Shows the name and thumbnail of the task.
+:::tip Update Thumbnail
+You can update the thumbnail by right-clicking the thumbnail and selecting Upload new thumbnail.
+:::
+- **Tags**: View and add tags.
+- **Status**: View and update status.
+- **Web Actions**: Access available web actions to launch applications directly via the AYON web server.
+:::info Web Actions
+Requires core version >= `0.4.4` and Launcher >= `1.1.0`.
+:::
```
Comment (Innders) [inline website/docs/artist_details_panel.md:30]: This is user/artist facing and they will not understand this, so I would remove it.

File: website/docs/artist_details_panel.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+- **Activity Feed**: More than just comments. Learn more in the [Activity Feed](artist_activity_feed.md) section.
+- **Watchers**: Update notification settings for the current task and choose who receives updates.
```
Comment (Innders) [inline website/docs/artist_details_panel.md:32]:
```suggestion
- **[Activity Feed](artist_activity_feed.md)**: Leave comments and track activity.
```

File: website/docs/artist_details_panel.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+- **Picture in Picture**
+- **Escape**: Close the current details panel.
```
Comment (Innders) [inline website/docs/artist_details_panel.md:34]:
```suggestion
- **[Watchers](artist_inbox.md#watchers)**: Update notification settings for the current folder or task and choose who receives updates.
- **Picture in Picture**: Open a panel that can stay open as you navigate to different pages.
```

File: website/docs/artist_details_panel.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+When using the AYON production tracker, you'll encounter the details panel frequently. It's essential for team interactions.
+As the name suggests, the details panel is packed with information about tasks, serving as a communication tool and offering useful actions.
+
+
+## Panel Content
+
+![](assets/details_panel/details_panel_overview.png)
+
+- **Folder Path**: Displays the folder path of the current task.
+- **Name & Thumbnail**: Shows the name and thumbnail of the task.
```
Comment (Innders) [inline website/docs/artist_details_panel.md:21]:
```suggestion
- **Folder Path**: Displays the folder path to the entity.
```

File: website/docs/artist_details_panel.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+:::tip Update Thumbnail
```
Comment (Innders) [inline website/docs/artist_details_panel.md:22]:
```suggestion
- **Name & Thumbnail**: Shows the name and thumbnail of the entity.
```

File: website/docs/artist_details_panel.md [hunk 11]
```diff
@@ -0,0 +1,40 @@
+---
+id: artist_details_panel
+title: The Task Details Panel
+sidebar_label: Details Panel
```
Comment (Innders) [inline website/docs/artist_details_panel.md:4]:
```suggestion
title: The Details Panel
```

The details panel is not just for the task but for all entities.

File: website/docs/artist_details_panel.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+:::info
+The details panels received updates in server version `1.3.0`.
+:::
+
+## Overview 
+When using the AYON production tracker, you'll encounter the details panel frequently. It's essential for team interactions.
+As the name suggests, the details panel is packed with information about tasks, serving as a communication tool and offering useful actions.
+
```
Comment (Innders) [inline website/docs/artist_details_panel.md:14]:
```suggestion
As the name suggests, the details panel is packed with information, serving as a communication tool and offering useful actions.
```

**Review by Innders — CHANGES_REQUESTED** [review]

File: website/docs/artist_details_panel.md (hunk 12 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md:10]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 11 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md:4]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 10 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md:18]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 9 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 8 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md:27]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 7 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 6 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 5 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/addon_applications_admin.md (hunk 4 above)
Comment (MustafaJafar) [reply website/docs/addon_applications_admin.md:50]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/addon_applications_admin.md (hunk 2 above)
Comment (MustafaJafar) [reply website/docs/addon_applications_admin.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

File: website/docs/artist_details_panel.md (hunk 1 above)
Comment (MustafaJafar) [reply website/docs/artist_details_panel.md]: Thank you. 
https://github.com/ynput/ayon-documentation/pull/287/commits/069be35feeac7fabad542741b1bfa6d7643d8195

**Review by Innders — APPROVED** [review]
Awesome stuff 💯

File: website/docs/addon_applications_admin.md [hunk 13]
```diff
@@ -30,6 +33,22 @@ The Addon features two primary settings categories: Definitions and Filters:
...
+:::info
+The web actions feature requires AYON Server version `1.3.0` or higher, core addon version `0.4.4` or higher, and Launcher version `1.1.0` or higher.
+
+Web actions run through **shims**, which are automatically set up when users install the AYON launcher version `1.1.0` or higher.
+:::
+
+Web actions let you launch applications directly through the AYON web server. You can find these actions in the [details panel](artist_details_panel.md).
+
+The application addon handles the logic for web actions.Currently, there are no specific settings for web actions.
```
Comment (BigRoy) [inline website/docs/addon_applications_admin.md:47]:
```suggestion
The application addon handles the logic for web actions. Currently, there are no specific settings for web actions.
```

File: website/docs/addon_applications_admin.md [hunk 14]
```diff
@@ -30,6 +33,22 @@ The Addon features two primary settings categories: Definitions and Filters:
...
+
+Web actions are generated from your application's definitions in your production or development bundles, depending on the developer mode setting, and they adhere to your application’s filter settings.
```
Comment (BigRoy) [inline website/docs/addon_applications_admin.md:49]:
```suggestion
The application web actions are generated from your application's definitions in your production or development bundles, depending on the developer mode setting, and they adhere to your application’s filter settings.
```
I think it may be nice to be extra clear here once more that we're only talking about the web actions for applications.

File: website/docs/dev_addon_creation.md [hunk 15]
```diff
@@ -275,10 +275,168 @@ If you are curious, here's what happens when you access the api without logging
...
 
 ### Web Action
-:::note TODO
-Write about it and add an example.
+
+Web actions let users trigger AYON Addons CLI actions directly from the AYON server to perform actions on their machines. They run through **shims**, which are automatically set up when users install the AYON launcher version `1.1.0` or higher.
+
+Web actions are added in two steps:
+
+1. **Client-Side CLI Actions**: Develop the addon CLI actions in the client-side code of the addon. More about them in the next section [CLI Interface](dev_addon_creation.md#cli-interface).
```
Comment (BigRoy) [inline website/docs/dev_addon_creation.md:283]: @iLLiCiTiT I vaguely recall you saying that _technically_ it's not a requirement for the web action to go through the launcher CLI - it could potentially do something on the machine without launcher or CLI being involved? Did I misunderstand?

File: website/docs/dev_addon_creation.md [hunk 16]
```diff
@@ -275,10 +275,168 @@ If you are curious, here's what happens when you access the api without logging
...
+            "name": "switch_access_2",
+        }
+        output.append(
+            SimpleActionManifest(
+                identifier=f"{IDENTIFIER_PREFIX}.show_dialog",
+                label=label,
+                icon=icon,
+                order=100,
+                entity_type="folder",
+                entity_subtypes=None,
```
Comment (BigRoy) [inline website/docs/dev_addon_creation.md:388]: I recall - as I tried to implement these - being quite confused about what combinations of values this support. Should we document that clearer somewhere (either in the code somewhere, or here?)

File: website/docs/addon_applications_admin.md (hunk 13 above)
Comment (MustafaJafar) [reply website/docs/addon_applications_admin.md]: Thank you https://github.com/ynput/ayon-documentation/pull/287/commits/970a51c94d067dbd8e5c09e153f0b83387eb2fb2

File: website/docs/addon_applications_admin.md (hunk 14 above)
Comment (MustafaJafar) [reply website/docs/addon_applications_admin.md]: Thank you https://github.com/ynput/ayon-documentation/pull/287/commits/970a51c94d067dbd8e5c09e153f0b83387eb2fb2

File: website/docs/dev_addon_creation.md (hunk 15 above)
Comment (MustafaJafar) [reply website/docs/dev_addon_creation.md:283]: As far as I know, this is not implemented yet in the [executor](https://github.com/ynput/ayon-backend/blob/develop/ayon_server/actions/execute.py).
And, will we need to setup shims for other applications we want to run directly from the web action?

---
Personally, I like using launcher CLI as we can add the logic for different operating systems there instead of the web action.

Comment (BigRoy) [reply website/docs/dev_addon_creation.md:283]: Hmm, yes agreed.

File: website/docs/dev_addon_creation.md (hunk 16 above)
Comment (MustafaJafar) [reply website/docs/dev_addon_creation.md:388]: @martastain 
TL;DR 
This PR adds draft/initial docs for implementing web actions. Also, I've added an example web action. 
My question: Do you have any recommendations for extending the current initial docs?

Comment (BigRoy) [reply website/docs/dev_addon_creation.md:388]: I'm personally fine with it staying as is - just saying that it'd be nice to get this clarified over time because it's where I got lost.

**Comment by Innders** [comment]
@MustafaJafar this is never going to get approved by everyone, probably better to merge now and improve later?

**Comment by MustafaJafar** [comment]
> @MustafaJafar this is never going to get approved by everyone, probably better to merge now and improve later?

yeah, let's merge it.

File: website/docs/dev_addon_creation.md [hunk 17]
```diff
@@ -275,10 +275,168 @@ If you are curious, here's what happens when you access the api without logging
...
+<!-- TODO: We may expand later on `SimpleActionManifest` and `ActionExecutor`. -->
+
+<details><summary>Simple Web Action Example</summary>
+
+This simple example triggers a QT dialog on the user's machine, showing the folder path from which the action was triggered.
+
+![](assets/addon_dev/web_action_example.png)
+
+```python title="my_addon/client/my_addon/__init__.py"
+from qtpy import QtWidgets
```
Comment (iLLiCiTiT) [inline website/docs/dev_addon_creation.md:295]: It's too late, but Qt (or any other runtime dependency) should never be imported in addon...

Comment (iLLiCiTiT) [reply website/docs/dev_addon_creation.md:295]: > For reference, should it be imported inside the function like [here](https://github.com/ynput/example-studio-addon/blob/406ac8d56bb7ae3ed556859bd51b2081c79e8a8d/client/my_studio_addon/addon.py#L69)?

Yes, usually by importing Qt it in the function, or by importing file that has Qt implementation inside the function.
